### PR TITLE
JP-4372: Fix edge cases in bounding box computations

### PR DESCRIPTION
**Pull request by mcara**
Fixes a crash in the JWST pipeline which happens when one of the images has three sources detected and they lie on almost a straight line. The convex hull of these points is a triangle with minimal height resulting in the spherical polygon being considered degenerate since all vertices lie on a great circle.

This PR replaces convex hull with a bounding rectangle of height or with of no less than 1 pixel only when the spherical polygon created with the convex hull is degenerate.

Closes [JP-4372](https://jira.stsci.edu/browse/JP-4372)

Regression tests:
- JWST: https://github.com/spacetelescope/RegressionTests/actions/runs/27257813514
- JWST (`CRDS_CONTEXT=jwst_1555.pmap`): https://github.com/spacetelescope/RegressionTests/actions/runs/27310278928
- Roman: https://github.com/spacetelescope/RegressionTests/actions/runs/27257905170

### CODE REVIEW

**Comment by tapastro** [comment]
I can't say that I fully follow the new code - it seems like there are instances were 0.5 is added or removed from array values to generate some non-zero areas, but I don't understand what units x and y typically have here - pixel coordinates, or something else? It's not immediately clear to me that 0.5 is a small value. But I think the concept appears sound.

**Comment by mcara** [comment]
> I can't say that I fully follow the new code - it seems like there are instances were 0.5 is added or removed from array values to generate some non-zero areas, but I don't understand what units x and y typically have here - pixel coordinates, or something else? It's not immediately clear to me that 0.5 is a small value. But I think the concept appears sound.

@tapastro You are right that for the case of two sources there were two duplicate subtractions of half pixel. I fixed that in the latest commit. The units are pixels for image catalogs. The fix essentially attempts to build the minimum area box (with minimum width and height of 1 pixel) that encloses all the sources.

The issue that was causing the crash in the dataset that you shared is that one of the images had 3 stars arranged essentially along a straight line resulting in a bounding polygon (a triangle) of a very small height (I think <0.01 pixels). This was detected by the `spherical_geometry` as a degenerate polygon. The fix here tries to ensure that the bounding polygon is not degenerate (does not have vertices along a straight line).

File: tweakwcs/wcsimage.py [hunk 1]
```diff
@@ -500,14 +500,99 @@ def _calc_cat_convex_hull(self):
...
+            dx /= d
+            dy /= d
+            du = -dy
+            dv = dx
+            # build a rectangle by shifting the line segment by 0.5 in the
+            # direction perpendicular to the line:
+            x = np.array([x[0] + 0.5 * du, x[0] - 0.5 * du, x[1] - 0.5 * du, x[1] + 0.5 * du, x[0] + 0.5 * du])
+            y = np.array([y[0] + 0.5 * dv, y[0] - 0.5 * dv, y[1] - 0.5 * dv, y[1] + 0.5 * dv, y[0] + 0.5 * dv])
+
+        elif len(x) > 3:
```
Comment (melanieclarke) [inline tweakwcs/wcsimage.py]: I think this misses the case of len(x) == 3.  Is that intended?

File: CHANGELOG.rst [hunk 2]
```diff
@@ -5,6 +5,13 @@ Release Notes
 =============
 
 
+0.9.1 (unreleased)
+==================
+
+- Improved handling of edge cases in bounding poligon calculations for
```
Comment (melanieclarke) [inline CHANGELOG.rst]:
```suggestion
- Improved handling of edge cases in bounding polygon calculations for
```

**Comment by melanieclarke** [comment]
I tested this fix on the test data for jw07671-o057_image3_00001_asn.json.  I can reproduce the error with the latest release candidate environment (via stasis).  Installing this branch in that environment fixes the error and allows processing to complete.  Results with the fix look equivalent to tests with jwst on main in my development environment, which does not show the error.

File: tweakwcs/wcsimage.py (hunk 1 above)
Comment (mcara) [reply tweakwcs/wcsimage.py]: You are correct. Fixed in the last commit

**Comment by mcara** [comment]
All tests pass
